namespaces/resource-namespace.md

```diff
@@ -0,0 +1,231 @@
+    - `parameters` (Href Variables) - Input parameters for this transition.
+    - `attributes` (Data Structure) - Input attributes for this transition.
+- `content` (array[HTTP Transaction]) - Array of transaction examples.
+
+    Transaction examples are protocol-specific examples of REST transaction
+    that was initialized by exercising a transition.
+
+    For the time being this namespace defines only HTTP-specific transaction
+    data structures.
+
+#### Example
+
+```json
+{
+    "element": "transition",
+    "attributes": {
+        "relation": "update",
+        "href": "https://polls.apiblueprint.org/questions/{question_id}"
+    },
+    "content": null
+}
+```
+
+## Protocol-specific Elements
+
+### HTTP Transaction (Element)
+
+Example of an HTTP Transaction. The example's content consist of a reuqest-response
+message pair. A transaction example MUST contain exactly one HTTP request and one HTTP response message.
+
+#### Properties
+
+- `element`: `http-transaction` (string, fixed)
+- `content` (array, fixed) - Request response message pair (tuple).
+
+    The array MUST contain exactly two elements.
+
+    - Elements
+        - (HTTP Request Message)
+        - (HTTP Response Message)
+
+#### Example
+
+```json
+{
+    "element": "http-transaction",
+    "content": [
+        {
+            "element": "http-request",
+            "attributes": {
+                "method": "GET",
+                "href": "https://polls.apiblueprint.org/questions/1"
+            },
+            "content": null
+        },
+        {
+            "element": "http-response",
+            "attributes": {
+                "status": 200,
+                "assets": [
+                    {
+                        "element": "asset",
+                        "meta": {
+                            "class": "message-body"
+                        },
+                        "attributes": {},
+                        "content": "{\"name\": \"John\"}"
+                    }
+                ]
+            },
+            "content": null
+        }
+    ]
+}
+```
+
+### HTTP Message (Element)
+
+- `element`: `http-message` (string, fixed)
+- `attributes`
+    - `headers` (HTTP Headers) - Array of HTTP headers.
+    - `assets` (array[Asset]) - Array of assets associated with the message.
+- `content` (Data Structure)
+
+### HTTP Headers (Object Type)
+
+Hash of HTTP header-fields where keys are `field-name` as defined in
+[RFC 7230][].
+
+#### Properties
+
+- `element`: `http-headers`
+
+### HTTP Request Message (HTTP Message)
+
+- `element`: `http-request` (string, fixed)
+- `attributes`
+    - `method` (string) - HTTP request method.
+    - `href` (string) - A concrete URI for the request.
+
+        Note: Value of this property MUST not be an URI Template.
+
+### HTTP Response Message (HTTP Message)
+
+- `element`: `http-response` (string, fixed)
+- `attributes`
+    - `statusCode` (number) - HTTP response status code
```

Comment:
Do you think it is worth trying to capture responses that have multiple status codes, i.e. [HTTP 100 Continue](https://support.urbanairship.com/entries/59909909--Expect-100-Continue-Issues-and-Risks)? Urban Airship disabled the feature due to problems it was causing, but I know for a fact that Amazon S3 supports it and it's a commonly used pattern for APIs that handle large message bodies.

On the one hand I feel like it's an implementation detail not relevant to the higher level final request/response pair, on the other it seems important to include in the request/response examples so that implementers are aware of the feature.